sql/src/test/java/io/crate/analyze/InsertFromValuesAnalyzerTest.java

```diff
@@ -1011,28 +1011,35 @@ public void testOnDuplicateKeyUpdateOnObjectColumn() throws Exception {
     @Test
     public void testInvalidLeftSideExpressionInOnDuplicateKey() throws Exception {
         expectedException.expect(IllegalArgumentException.class);
-       e.analyze("insert into users (id, name) values (1, 'Arthur') on duplicate key update [1, 2] = 1");
+        e.analyze("insert into users (id, name) values (1, 'Arthur') on duplicate key update [1, 2] = 1");
     }
 
     @Test
     public void testUpdateOnPartitionedColumnShouldRaiseAnError() throws Exception {
         expectedException.expect(ColumnValidationException.class);
         expectedException.expectMessage("Validation failed for date: Updating a partitioned-by column is not supported");
-       e.analyze("insert into parted (id) values (1) on duplicate key update date = 0");
+        e.analyze("insert into parted (id) values (1) on duplicate key update date = 0");
     }
 
     @Test
-    public void testUpdateOnPrimaryKeyColumnShouldRaiseAnError() throws Exception {
+    public void testUpdateOnCompoundPrimaryKeyColumnShouldRaiseAnError() throws Exception {
         expectedException.expect(ColumnValidationException.class);
         expectedException.expectMessage("Validation failed for id: Updating a primary key is not supported");
-       e.analyze("insert into nested_pk (id, o) values (1, {b=1}) on duplicate key update id = 10");
+        e.analyze("insert into nested_pk (id, o) values (1, {b=1}) on duplicate key update id = 10");
+    }
+
+    @Test
+    public void testUpdateOnSinglePrimaryKeyColumnShouldRaiseAnError() throws Exception {
+        expectedException.expect(ColumnValidationException.class);
+        expectedException.expectMessage("Validation failed for id: Updating a primary key is not supported");
+        e.analyze("insert into users (id) values (1) on duplicate key update id = 10");
```

Comment:
why is that needed? afaik this case is covered by the new DocTableRelationTest.testEnsureNotUpdateExceptionMessagePrimaryKey

Reply:
then also testUpdateOnPartitionedColumnShouldRaiseAnError is obsolete

Reply:
In my opinion it makes more sense to keep these tests here than the manual call of valiation in DocTableRelationTest

Reply:
I prefer more encapsulated, explicit testing, but I'm ok to keep them in the analyzer test.